### PR TITLE
[CUMULUS-3147] - updating node version to 16.19

### PR DESCRIPTION
**Pull request by Nnaga1**
**Summary:** Summary of changes

Addresses [CUMULUS-XX: Develop amazing new feature](https://bugs.earthdata.nasa.gov/browse/CUMULUS-XXX)

## Changes

* Detailed list or prose of changes
* ...

## PR Checklist

- [ x] Update CHANGELOG
- [ ] Unit tests
- [ ] Ad-hoc testing - Deploy changes and test manually
- [ ] Integration tests

### CODE REVIEW

File: example/spec/parallel/s3_provider.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+    "id": "s3_provider",
+    "protocol": "s3",
+    "host": "pgc-cum-test-data-shared",
+    "globalConnectionLimit": 10
+}
```
Comment (jennyhliu) [inline example/spec/parallel/s3_provider.json:6]: What's this file for?

File: package.json [hunk 2]
```diff
@@ -76,6 +76,8 @@
     "@ava/typescript": "^1.1.1",
     "@babel/core": "^7.9.6",
     "@octokit/graphql": "2.1.1",
+    "@tsconfig/node16": "^1.0.3",
+    "@tsconfig/node16-strictest": "^1.0.3",
```
Comment (jennyhliu) [inline package.json:80]: What are these for?

File: packages/aws-client/src/S3.ts [hunk 3]
```diff
@@ -373,7 +373,7 @@ export const downloadS3File = async (
   const objectStream = await getObjectReadStream({
     bucket: s3Obj.Bucket,
     key: s3Obj.Key,
-    s3: s3(),
+    s3: Object.assign(s3()),
```
Comment (jennyhliu) [inline packages/aws-client/src/S3.ts:376]: why? Same question for the other Object.assign changes

File: tsconfig.json [hunk 4]
```diff
@@ -1,13 +1,20 @@
...
     "module": "commonjs",
     "esModuleInterop": true,
     "declaration": true,
     "declarationMap": true,
     "sourceMap": true,
     "removeComments": false,
     "strict": true,
+    "skipLibCheck": true,
+    "forceConsistentCasingInFileNames": true,
+    "moduleResolution": "node",
```
Comment (jennyhliu) [inline tsconfig.json:17]: Are these changes necessary for node16?  Do you have related document link?

**Review by jennyhliu — CHANGES_REQUESTED** [review]
1. We probably want to use the latest node 16
2. Update https://github.com/nasa/cumulus/pull/3228/files#diff-16668ffceabc29f1ac6035001afb94920ff2c385359e558fbffa51b94a8a0e27R6
We need to build image, publish and use it for our example stack build.
https://github.com/nasa/cumulus/tree/master/packages/api/ecs/async-operation
3. Publish a node 16 image for ci testing.
4. Add test to verify the update solve the current_buffer_.IsEmpty() issue daac has.

**Comment by jennyhliu** [comment]
> Is there documentation on how to do this? `Publish a node 16 image for ci testing.`
Nate and Jonathan know how to.

File: package.json (hunk 2 above)
Comment (jennyhliu) [reply package.json:80]: Do we need both "@tsconfig/node16" and "@tsconfig/node16-strictest"?

File: tsconfig.json (hunk 4 above)
Comment (jennyhliu) [reply tsconfig.json:17]: Should it be `"extends": "@tsconfig/node16-strictest/tsconfig.json"`?

Comment (jennyhliu) [reply tsconfig.json:17]: and https://www.npmjs.com/package/@tsconfig/node16-strictest

Comment (jennyhliu) [reply tsconfig.json:17]: I think only `"target": "es2021",` is necessary.  The other options change the current typescript setting. @Jkovarik any suggestion?

File: package.json (hunk 2 above)
Comment (jennyhliu) [reply package.json:80]: why didn't we have @tsconfig/node14 as dependency?

Comment (jennyhliu) [reply package.json:80]: @Jkovarik do you know why we don't need `"@tsconfig/node14` and if it's necessary to add `"@tsconfig/node16-strictest`?

**Comment by jennyhliu** [comment]
> Also quick question, this link in # 2 of your suggestions, https://github.com/nasa/cumulus/pull/3228/files#diff-16668ffceabc29f1ac6035001afb94920ff2c385359e558fbffa51b94a8a0e27R6, I'm not exactly clear on what file to update (or was this an extension of step 1 to bump 16.17 to 16.19?)

Wrong link, update the Dockerfile 
https://github.com/nasa/cumulus/blob/master/packages/api/ecs/async-operation/Dockerfile#L1

File: package.json (hunk 2 above)
Comment (Jkovarik) [reply package.json:80]: @jennyhliu  I don't know off the top of my head -- I think the AC on this ticket was talking about updating lines like 

```
    "target": "ES2019",
```
in the tsconfig.json.    I'm perhaps misunderstanding the question though.

Comment (Jkovarik) [reply package.json:80]: I'm assuming this was related to the arch meeting discussion where utilizing a common ts configuration was discussed?

**Review by marchuffnagle — CHANGES_REQUESTED** [review]

**Comment by marchuffnagle** [comment]
@Nnaga1 About the tsconfig changes:

Using the tsconfig bases is generally the right approach, rather than having to figure out the best config yourself. There are a couple of options there, usually something like "16" and "16-strictest". Using the strictest config is usually better, that identifies more issues than the more permissive "16" config would.

The easiest way to use those base configs is to add the package to `package.json` with `npm install --save-dev @tsconfig/node16-strictest`, then create a `tsconfig.json` file that is simply:

```json
{
  "extends": "@tsconfig/node16-strictest/tsconfig.json"
}
```

That will use the tsconfig settings that are defined in that package. The downside to that approach is that it isn't clear _what_ those settings are. My preference is to copy and paste the full tsconfig into our tsconfig file, and add a comment showing where it came from. So, in this case, it would start with:

```json
// Copied from https://raw.githubusercontent.com/tsconfig/bases/main/bases/node16-strictest.combined.json
{
  "$schema": "https://json.schemastore.org/tsconfig",
  "display": "Node 16 + Strictest",
  "compilerOptions": {
 ...
  rest of config here
```

If you take this approach (which you should), then you don't need to add the tsconfig base packages to package.json.

Comparing that `node16-strictest` config with the config were using for Node 14, it looks like there are a couple of settings that we'll want to specify in that config. `"declaration": true`, etc. If you add those to tsconfig, though, then it's hard to tell what came from the recommended config and what we are customizing. The way I like to handle that is to save that recommended config into a `tsconfig.base.json` file, and then have a `tsconfig.json` file that extends that base config. So, in our case, our `tsconfig.json` file would look like:

```json
{
  "extends": "./tsconfig.base.json",
  "compilerOptions": {
    "declaration": true,
    "declarationMap": true,
    "sourceMap": true,
    "removeComments": false,
    "incremental": true
  }
}
```

File: docs/workflows/lambda.md [hunk 5]
```diff
@@ -10,7 +10,7 @@ AWS provides great getting started guide for building Lambdas in the [developer
 
 Cumulus currently supports the following environments for Cumulus Message Adapter enabled functions:
 
-* [Node.js 12.18 - 14.19](https://docs.aws.amazon.com/lambda/latest/dg/programming-model.html)
+* [Node.js 12.18 - 16.19](https://docs.aws.amazon.com/lambda/latest/dg/programming-model.html)
```
[the remark this answers is not in the document]
Comment (marchuffnagle) [reply docs/workflows/lambda.md:13]: I'll defer to someone like @Jkovarik

File: package.json (hunk 2 above)
Comment (jennyhliu) [reply package.json:79]: There are other places in the baseline have "target": "es2019" and "target": "ES2020", please check if we need to update those.

File: package.json [hunk 6]
```diff
@@ -76,6 +76,7 @@
     "@ava/typescript": "^1.1.1",
     "@babel/core": "^7.9.6",
     "@octokit/graphql": "2.1.1",
+    "@tsconfig/node16-strictest": "^1.0.4",
```
Comment (marchuffnagle) [inline package.json:79]: Based on [this comment](https://github.com/nasa/cumulus/pull/3228#issuecomment-1405233837), this is not needed and can be removed.

File: CHANGELOG.md [hunk 7]
```diff
@@ -88,7 +88,8 @@ update the database cluster to use the new configuration.
   - Fixed `sqsMessageRemover` lambda to correctly retrieve ENABLED sqs rules.
```
Comment (jennyhliu) [inline CHANGELOG.md:89]: Need to add `Breaking Changes` and `Notable changes` for this ticket, add docker images information
Similar to [this](https://github.com/nasa/cumulus/blob/master/CHANGELOG.md#v1118-2022-11-07-backport)

Comment (jennyhliu) [reply CHANGELOG.md:89]: Update tf-modules/cumulus_ecs_service/README.md docs/data-cookbooks/run-tasks-in-lambda-or-docker.md to new image version.

File: .vscode.example/launch.json [hunk 8]
```diff
@@ -57,7 +57,7 @@
       "args": [
         "${file}"
       ],
```
Comment (jennyhliu) [inline .vscode.example/launch.json:59]: line 43 has `"runtimeVersion": "14.19.1",`

**Review by jennyhliu — APPROVED** [review]
nice job!
